modules/services/signaturepdf.nix

```
@@ -0,0 +1,68 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.signaturepdf;
+  extraConfigToArgs = extraConfig:
+    lib.flatten
+    (lib.mapAttrsToList (name: value: [ "-d" "${name}=${value}" ]) extraConfig);
+in {
+  meta.maintainers = [ lib.maintainers.DamienCassou ];
+
+  options.services.signaturepdf = with lib; {
+    enable = mkEnableOption "signaturepdf, the clipboard management daemon";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.signaturepdf;
+      defaultText = "pkgs.signaturepdf";
+      description = "signaturepdf derivation to use.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8080;
+      example = 8081;
+      description = "The port on which the application runs";
+    };
+
+    extraConfig = mkOption {
```

I guess this should be 
```suggestion
    settings = mkOption {
```
and the type probably should be something like
```nix
with types;
  let primitive = oneOf [ str int bool float ];
  in attrsOf primitive;
```

---

good idea, done.